src/fundamental/macro-fundamental.h

```diff
@@ -249,6 +249,30 @@
                         CONST_ISPOWEROF2(_x);                          \
                 }))
 
+#define ADD_SAFE(ret, a, b) (!__builtin_add_overflow((a), (b), (ret)))
```

Comment:
hmm, i think it would make sense do change this to:

```c
#define ADD_SAFE(ret, a, b) (!__builtin_add_overflow((a), (b), &(ret)))
```

i.e. automatically do the `&` here, since we'll need that always, no?

Reply:
> since we'll need that always

No. In fact I even use it without the & in `homework-blob`. I already have an int* that I pass into the macro raw

I think it's easier to reason about macros when they behave like functions. That way you don't have to double check what the macro expands to so you don't accidentally dereference or neglect to dereference something. For example, the `free_and_replace` macros really confuse me and I always have to check whether or not they imply an address-of operator (I think they do, right?)